src/enums.h

```diff
@@ -363,4 +363,26 @@ std::array<T, N> vec_to_array(const std::vector<T> &vec)
     return array;
 }
 
+template <typename T>
```

Comment:
after I had just responded to Narc too... ;-)
just because it's you @BevapDin, I will change it over. :-P